backend-processing.ts

```diff
@@ -96,24 +99,65 @@ export default class BackendProcessing {
     }
     let url = entry["hyperlink url"];
     let comments = entry["comments"];
-    database.insertFromCsvLine(
-      userID,
-      fundID,
-      firstCompany,
-      name,
-      firstPosition,
-      sterm,
-      eterm,
-      employer,
-      position,
-      url,
-      comments
-    );
+    database
+      .insertFromCsvLine(userID, fundID, firstCompany, name, firstPosition, sterm, eterm, employer, position, url, comments)
+      .then((individualID) => {
+        let q1 = entry["q1 " + year];
+        let q2 = entry["q2 " + year];
+        let q3 = entry["q3 " + year];
+        let q4 = entry["q4 " + year];
+        this.processQuarter(q1, 1, individualID, userID, fundID, year);
+        this.processQuarter(q2, 2, individualID, userID, fundID, year);
+        this.processQuarter(q3, 3, individualID, userID, fundID, year);
+        this.processQuarter(q4, 4, individualID, userID, fundID, year);
+      });
+  }
+  processQuarter(quarterString, quarterNumber, individualID, userID, fundID, year) {
+    if (quarterString == null || quarterString == "") return;
+    let quarterDates = [year + "/01/01", year + "/04/01", year + "/07/01", year + "/10/01"];
+    let positionName = "";
+    let companyName = "";
+    let p = quarterString.indexOf(" at ");
+    if (p > 0) {
+      positionName = quarterString.substring(0, p);
+      companyName = quarterString.substring(p + 4);
+    }
+    p = quarterString.indexOf(",");
+    if (p > 0) {
+      positionName = quarterString.substring(0, p);
+      companyName = quarterString.substring(p + 1);
+    }
+    companyName = companyName.trim();
+    if (positionName == "" || companyName == "") return;
+    database.insertQuarterEmployment(userID, individualID, fundID, companyName, positionName, quarterDates[quarterNumber - 1]);
+  }
+
+  //Attempts to find year numbers in the file, if it cannot recognize a year number then it defaults to the current year.
+  estimateYear(line: string): number {
+    line = line + "xxxx";
+    let i = line.indexOf("Q1 20");
```

Comment:
**IMPORTANT:** I would recommend asking the clients if all of their CSV's have this "Q1 20" format, or if it's just the ones for the year 2020. After all, should be able to cover all the years possible in the 20XX range, including any 2021's. If I have misread any part of this function that covers that, please carry on.